python/src/nanoarrow/_lib.pyx

```diff
@@ -1066,10 +1068,15 @@ cdef class CArray:
         return out
 
     def __getitem__(self, k):
+        self._assert_valid()
+
+        cdef int64_t kint
+
         if not isinstance(k, slice):
-            raise TypeError(
-                f"Can't slice CArray with object of type {type(k).__name__}"
-            )
+            kint = k
+            if kint < 0:
+                kint += self._ptr.length
+            return CScalar(self, kint)
```

Comment:
Is this needed for CArray? We could also leave this for just the higher-level Array object to give scalars?